torch/_higher_order_ops/triton_kernel_wrap.py

```diff
@@ -1231,12 +1303,100 @@ def call_triton_kernel(
                             updated = True
 
                 if updated:
-                    new_kernel = autotune(configs=new_configs, key=[])(
-                        variable.kernel.fn
-                    )
+                    prune_configs_by = {
+                        "perf_model": variable.kernel.perf_model,
+                        "early_config_prune": variable.kernel.early_config_prune,
+                        "configs_top_k": variable.kernel.configs_top_k,
+                    }
+
+                    new_kernel = autotune(
+                        configs=new_configs, prune_configs_by=prune_configs_by, key=[]
+                    )(variable.kernel.fn)
                     new_var = type(variable)(new_kernel, None, variable.grid)
                     return self.call_triton_kernel(new_var, args, kwargs, tx)
 
+        if "num_ctas" in kwargs:
+            self.raise_unsupported(
+                "Passing num_ctas directly to the Triton kernel is not supported. "
+                "Please use a Config in @triton.autotune instead."
+            )
```

Comment:
the answer to this was kind of interesting.

I had to set variable.source in call_getitem in functions.py in order to pass the source through (we need a valid base source for AttrSource). This is because of how Triton kernels are called with the grid.

It turns out that passing the source through breaks that test, and causes it to fail with a runtime error:  

<img width="1543" alt="image" src="https://github.com/user-attachments/assets/868bcc9f-000f-4432-bf51-50dd23e775bd" />

Moving the check down fixes the test, but I'll think about this some more and see if there is a better way to address this